### ISSUE TITLE
Scheduler hits floating point exception (dividing by 0)

### ISSUE THREAD

**Issue by jjsjann123** [csarofeen/pytorch#1466]
### 🐛 Describe the bug

I'm hitting issue when we have size-0 rank 1 tensors going through a reduction kernel.

```
import torch                                                                                                            
                                                                                                                        
@torch.jit.script                                                                                                       
def f(t):                                                                                                               
    return torch.nn.functional.softmax(t, dim=-1)                                                                       
                                                                                                                        
t0 = torch.randn(0, device="cuda")                                                                                      
s0 = torch.tensor(1.0, device="cuda")                                                                                   
                                                                                                                                                                                                                   
torch._C._jit_set_nvfuser_single_node_mode(True)                                                                        
                                                                                                                        
with torch.jit.fuser("fuser2"):                                                                                         
    f(t0)                                                                                                               
    f(t0)                                                                                                               
    f(t0)                                                                                                               
    print(f.graph_for(t0)) 
```

I tried to patch it with gdb but that just exposed more divide by 0 :fearful: 

### Versions

Reproed on top of devel

```
commit 592fd4762a8004c71d4acd5d885e04a85ccceae0 (csarofeen/devel)
Author: jjsjann123 <jiej@nvidia.com>
Date:   Fri Feb 11 23:52:03 2022 -0800

    disabling 0-dim cuda tensor reduction/normalization (#1453)
    
    1. Disables normalizatoin/reduction for 0-dim cuda inputs.
    2. Patches fusion logic on reduction parsing rule
    
    We are doing this because normalization for 0-dim behavior is tricky to justify (?!) and seems implementation specific. So we just leave this for eager.

```

**Comment by jjsjann123** [comment]
cc'ing @csarofeen

**Comment by jjsjann123** [comment]
It's unfortunate that we can't easily exclude size-0 input tensors since we don't specialize them at the time. At least our PW scheduling/kernel works fine with size-0 dimensions, so I tend to think this should also work for normalizations?!

If that's something that we want to do, we can revisit the guard logic.
